app/scripts/app.js

```diff
@@ -38,4 +50,15 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
     }
   };
 
+  app.persistChallenge = function(challenge){
+    app.challenge = challenge;
+    document.cookie = "challengeId="+challenge.id;
```

Comment:
we will add an expiring date to the challengeId so the user cannot access that challenge after some time so in my opinion the cookies are the best storage option.